crypto/x509/x509_vfy.c

```diff
@@ -2980,7 +2984,14 @@ static int build_chain(X509_STORE_CTX *ctx)
                  * certificate among the ones from the trust store.
                  */
                 if ((search & S_DOALTERNATE) != 0) {
-                    OPENSSL_assert(num > i && i > 0 && ss == 0);
+                    if (!ossl_assert(num > i && i > 0 && ss == 0)) {
+                        X509err(X509_F_BUILD_CHAIN, ERR_R_INTERNAL_ERROR);
+                        X509_free(xtmp);
+                        trust = X509_TRUST_REJECTED;
+                        ctx->error = X509_V_ERR_UNSPECIFIED;
+                        search = 0;
+                        continue;
```

Comment:
WRT `break` I'd prefer not to use a different pattern to that used elsewhere (that in itself would be confusing I think)...and I'm not keen on changing all the other occurrences as part of this PR.

WRT `/* last change to make a trusted chain */`, I don't think it can be moved into the switch. That block updates the value of `trust` which is then used by the subsequent switch. I can't see why the above altered flow would make it inappropriate to call `sk_X509_num(ctx->chain)`...no changes are made in that flow to the chain. *Possibly* you could wrap the whole block in an `if (trust == X509_TRUST_UNTRUSTED)` if you really think it is necessary. That would avoid the call. I'm not really convinced it is necessary.